packages/react-ui-core/src/DatePicker/DatePicker.js

```diff
@@ -33,7 +34,7 @@ export default class DatePicker extends PureComponent {
 
     this.picker = React.createRef()
     this.outsideHandler = null
-    this.formatRegex = new RegExp(props.dateFormat.replace('/', '\\/').replace(/[mdy]/ig, '\\d'))
```

Comment:
This wasn't actually creating a valid regex. Decided to hardcode what it was attempting to validate. Which is `mm/dd/yyyy`